opencog/atoms/proto/ValueFactory.h

```diff
@@ -0,0 +1,98 @@
+#ifndef _VALUE_FACTORY_H_
+#define _VALUE_FACTORY_H_
+
+#include "ProtoAtom.h"
+#include <opencog/atoms/base/Handle.h>
+
+#include <exception>
+#include <map>
+#include <typeinfo>
+#include <typeindex>
+#include <vector>
+
+namespace opencog
+{
+using CreateProto = ProtoAtomPtr (*) (...);
+
+struct FuncRegister
+{
+    CreateProto func;
+    std::vector<std::type_index> args;
+};
```

Comment:
`struct FuncRegister` should be cleared frivate, inside of `class ValueFactory` below.